src/main/java/com/google/sps/data/EventUpdater.java

```diff
@@ -36,7 +49,159 @@ public Entity getEntity() {
   public void updateEvent(HttpServletRequest request, GivrUser user, boolean forRegistration, EmbeddedEntity historyUpdate) throws IllegalArgumentException {
-    
+    HashMap<String, String> formProperties = new HashMap<String, String>();
+
+    // Format is (Form EntryName, Entity PropertyName)
+    formProperties.put("event-primary-organization-id", "eventOwnerOrgId");
+    formProperties.put("event-name", "eventName");
+    formProperties.put("event-partner", "eventPartnerNames");
+    formProperties.put("event-details", "eventDetails");
+    formProperties.put("event-contact-email", "eventContactEmail");
+    formProperties.put("event-contact-phone-num", "eventContactPhone");
+    formProperties.put("event-contact-name", "eventContactName");
+    formProperties.put("event-street-address", "eventStreetAddress");
+    formProperties.put("event-city", "eventCity");
+    formProperties.put("event-state", "eventState");
+    formProperties.put("event-zip-code", "eventZipcode");
+
+    /* Optional Properties can be left blank in the request form.
+     *
+     * The following properties are optional:
+     * - eventPartnerNames
+     * - eventDetails
+     */
+    Set<String> optionalProperties = new HashSet<String>();
+    optionalProperties.add("eventPartnerNames");
+    optionalProperties.add("eventDetails");
+
+    long ownerOrgId = 0;
+
+    for (Map.Entry<String, String> entry: formProperties.entrySet()) {
+      String propertyKey = entry.getValue();
+      String formKey = entry.getKey();
+      String formValue = "";
+
+      if (optionalProperties.contains(propertyKey)) {
+        formValue = request.getParameter(formKey) == null ? "" : request.getParameter(formKey);
+      } else {
+        try {
+          formValue = getParameterOrThrow(request, formKey);
+        } catch (IllegalArgumentException err) {
+          logger.log(Level.SEVERE, "Form value for: " + propertyKey + " cannot be left blank.");
+        }
+
+        if (formKey.equals("event-primary-organization-id")) {
+          ownerOrgId = Long.parseLong(formValue);
+        }
+      }
+
+      setEventProperty(propertyKey, formValue);
+    }
+
+    setNonFormProperties(forRegistration, historyUpdate);
+
+    setEventDateAndHours(request);
+
+    setEventOwnerOrgNameBasedOnId(ownerOrgId);
+  }
+
+  private void setEventOwnerOrgNameBasedOnId(long ownerOrgId) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    Filter queryFilter = new FilterPredicate(Entity.KEY_RESERVED_PROPERTY, FilterOperator.EQUAL, ownerOrgId);
+    Query query = new Query("Distributor").setFilter(queryFilter);
+    PreparedQuery preparedQuery = datastore.prepare(query);
+
+    Entity entity = null;
+    try {
+      entity = preparedQuery.asSingleEntity();
+    } catch(PreparedQuery.TooManyResultsException exception) {
+      logger.log(Level.SEVERE, "Multiple Distributor entities found with ID: " + Long.toString(ownerOrgId) + ".");
```

Comment:
Anywhere there is an exception, you should bubble it up to the servlet. Otherwise this will look like the update succeeded.